src/validations/rules/Writing_FedRAMP_Validations.md

```diff
@@ -0,0 +1,60 @@
+# How to create FedRAMP OSCAL validation constraints
+
+FedRAMP OSCAL documents are structured XML documents. The document syntax is defined by OSCAL XML Schema. However, XML Schema cannot express many semantic constraints, and an additional schema language is employed: Schematron.
+
+# Technologies used
+
+- [XML](https://www.w3.org/standards/xml/) is a structered document langauge.
+- [Namespaces in XML](https://www.w3.org/TR/xml-names/) are freqently encountered in structured XML documents
+- [OSCAL](https://pages.nist.gov/OSCAL/) is used to define the several FedRAMP document types in a structured data form. NIST provides XML Schema documents for the various OSCAL document types.
+- [Schematron](https://schematron.com/) is used to declare assertions for documents to be validated
+- [XPath](https://www.w3.org/TR/xpath-31/) is used in Schematron to define rule contexts and assertion tests
+- [XSpec](XSpec) is used to creaste unit tests for Schematron assertions
+- [XSLT](https://www.w3.org/TR/xslt-30/) is used to manipulate XML documents, and is used to transpile Schematron documents into XSLT documents
+
+# FedRAMP OSCAL References
+
+FedRAMP has adopted and extended NIST OSCAL document types. These adaptations are documented in the GSA [fedramp_automation repository](https://github.com/GSA/fedramp-automation/tree/master/documents).
+
+The pertinent normative documents are
+
+-[Guide to OSCAL-based FedRAMP Content](./Guide_to_OSCAL-based_FedRAMP_Content.pdf)
+
+-[Guide to OSCAL-based FedRAMP System Security Plans (SSP)](./Guide_to_OSCAL-based_FedRAMP_System_Security_Plans_(SSP).pdf)
+
+-[Guide to OSCAL-based FedRAMP Security Assessment Plans  (SAP)](./Guide_to_OSCAL-based_FedRAMP_Security_Assessment_Plans_(SAP).pdf)
+
+-[Guide to OSCAL-based FedRAMP Security Assessment Reports (SAR)](./Guide_to_OSCAL-based_FedRAMP_Security_Assessment_Reports_(SAR).pdf)
+
+-[Guide to OSCAL-based FedRAMP Plan of Action and Milestones (POA&M)](./Guide_to_OSCAL-based_FedRAMP_Plan_of_Action_and_Milestones_(POAM).pdf)
+
+Each describes how FedRAMP expects each OSCAL document type is to be presented in an authorization package.
+
+# FedRAMP validation constraints
+
+FedRAMP validation constraints extend the NIST OSCAL XML Schema constraints. The XML Schema constraints are essentially syntactic; Schematron is used to augment the former with semantic constraints which cannot be expressed using XML Schema.
+
+Schematron constraints have thus far been for system security plan documents. They are found in a Schematron document `ssp.sch`.
+
+A companion document `fedramp_values.xml` defines sets of allowed values for various FedRAMP OSCAL XML  elements and their attributes.
```

Comment:
Why do we do that?

> Constraints define required or recommended data elements FedRAMP expects in syntactically valid OSCAL documents. When developing rules, developers will recognize in most cases there are two pieces to a rule checking the constraint: the analytical query of the OSCAL data, and cross-referencing that data with a set of known values for given data elements in a constraint. Much like other software projects, FedRAMP has opted to store the known values as external configuration that can be adapted with changing the analytical query operation of the rule. These are the FedRAMP values.

Reply:
Adopted in https://github.com/18F/fedramp-automation/commit/bd71afaefcda11214aac1b0a6fd84d7f3e9a7762